packages/components/src/unit-control/index.tsx

```diff
@@ -168,6 +171,50 @@ function UnforwardedUnitControl(
 				refInputSuffix.current?.focus();
 			}
 		};
+		onPaste = ( event: ClipboardEvent< HTMLInputElement > ) => {
+			props.onPaste?.( event );
+			const pastedText = event.clipboardData.getData( 'text' );
+			const parsed = parseQuantityAndUnitFromRawValue(
+				pastedText,
+				units
+			);
```

Comment:
Do we check at all if the pasted unit is in the list of allowed units?

And what happens if it's not an allowed unit? Maybe we should adopt the [same strategy used to parse the `value`](https://github.com/WordPress/gutenberg/blob/744b298e63e9a3030fb350e0f2a760a37ac355c8/packages/components/src/unit-control/index.tsx#L78-L82) ?